src/Microsoft.Fx.Portability/ObjectModel/DiagnosticAnalyzerInfo.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System;
+
+namespace Microsoft.Fx.Portability.ObjectModel
+{
+#if FEATURE_SERIALIZABLE
+    [Serializable]
+#endif
+    public class DiagnosticAnalyzerInfo
+    {
+        public string AnalyzerName { get; set; }
+        public string Id { get; set; }
+        public ICollection<CompatibilityRangeAttribute> Attributes { get; set; }
+        public bool IsCompatibilityDiagnostic { get; set; }
+
+        /// <summary>
+        /// Returns the integer number value from the Id string property or a -1 if an integer value cannot be returned. 
+        /// </summary>
+        public int GetIdNumber()
+        {
+            if (string.IsNullOrEmpty(Id))
+            {
+                return -1;
+            }
+
+            var matches = Regex.Match(Id, @"(\d+)", RegexOptions.Compiled);
```

Comment:
Ah, good catch. I didn't realize they were cached (I suppose that makes sense). Given that, this matters less. It may still be nice (and a bit safer) to be explicit about the scope, though.